salt.loader: add error logging when whitelist lookup fails

terminalmage opened the pull request:
Rather than just raising a bare KeyError with no additional information, this raises a proper KeyError with the key that failed to be looked up. It also logs the key that failed to load, as well as the whitelist, to aid in troubleshooting.

Additionally, this resolves failed lazydict lookups in the tests due to improperly-set whitelists.

terminalmage committed:
Add error logging when whitelist lookup fails

Rather than just raising a bare KeyError with no additional information,
this raises a proper KeyError with the key that failed to be looked up.
It also logs the key that failed to load, as well as the whitelist, to
aid in troubleshooting.

terminalmage committed:
Fix loader whitelists in unit tests

This resolves failed lazydict lookups in the tests due to improperly-set
whitelists.

isbm commented on salt/loader.py:
```diff
@@ -1629,7 +1629,11 @@ def _load(self, key):
             if self.whitelist and mod_name not in self.whitelist:
-                raise KeyError
+                log.error(
+                    'Failed to load key %s because its module (%s) is not in '
```
If someone reading the log won't start panicking about SSH pr SSL keys... Can we rephrase this to something else?

terminalmage replied:
Sure, that is a good point.

terminalmage replied:
Done.

rallytime commented:
@terminalmage The dockermod tests are failing on this change. Can you take a look?

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-cent7-py3/5009/

terminalmage committed:
Change key -> function to make log message more clear

Leaving it as "key" would potentially confuse people into thinking there
was a problem with an SSH/SSL/TLS/etc. key.

terminalmage committed:
Add additional missing modules to whitelist

terminalmage commented:
@rallytime Should be fixed now. I though I ran all the dockermod unit tests after editing the whitelist, but it looks like I only ran a single test.
